### PR TITLE
New Update User Guide - Geoprocessing Tools section 

### PR DESCRIPTION
**Pull request by ElenaGallo**
## Description

Add the [Geoprocessing Tools](https://mapstore--9496.org.readthedocs.build/en/9496/user-guide/geoprocessing-tool/) section

## Other useful information
Related to #9343

### CODE REVIEW

File: docs/user-guide/geoprocessing-tool.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+By clicking the **GeoProcessing Tool** <img src="../img/button/geoprocessing-button.jpg" class="ms-docbutton"/> button, available in the [Side Toolbar](mapstore-toolbars.md#side-toolbar), a panel opens so that the user can choose the geographic operations between **Buffer** and **Intersection**.
+
+<img src="../img/geoprocessing-tool/geoprocessing-panel.jpg" class="ms-docimage" width="400px"/>
+
+## Buffer tool
+
+The **Buffer** tool allows to create a buffer around the input geometries and when it is selected, the user can:
+
+* Select a layer from the **Source Layer** option drop down menu. The dropdown will show the layers available for the process.
```
Comment (tdipisa) [inline docs/user-guide/geoprocessing-tool.md:16]:
```suggestion
* Select a layer from the **Source Layer** option drop down menu. The dropdown shows the layers available for the process from the ones present in TOC
```

File: docs/user-guide/geoprocessing-tool.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+* Select one of the layer's features from the **Source Feature**. The feature can be selected directly by the dropdown menu or from the map, activating the <img src="../img/button/add_marker_button.jpg" class="ms-docbutton"/> button.
```
Comment (tdipisa) [inline docs/user-guide/geoprocessing-tool.md:18]:
```suggestion
* Select one of the layer features from the **Source Feature** option. The feature can be selected from the dropdown menu or directly clicking on map by activating the <img src="../img/button/add_marker_button.jpg" class="ms-docbutton"/> button.
```

File: docs/user-guide/geoprocessing-tool.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+<video class="ms-docimage" controls><source src="../img/geoprocessing-tool/select-buffer-feature.mp4"/></video>
+
+* Insert the desired **Distance** for the buffer.
+
+The editor can now click <img src="../img/button/run_button.jpg" class="ms-docbutton"/> to start the process and generate the new buffer layer.
```
Comment (tdipisa) [inline docs/user-guide/geoprocessing-tool.md:24]:
```suggestion
When all mandatory process parameters have been provided, it is possible to click on  <img src="../img/button/run_button.jpg" class="ms-docbutton"/> button to start the process to generate and visualize the buffer layer.
```

File: docs/user-guide/geoprocessing-tool.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+<video class="ms-docimage" controls><source src="../img/geoprocessing-tool/run_buffer-layer.mp4"/></video>
+
+The buffer layer is now present in the [TOC](toc.md) and visible in the map viewer.
```
Comment (tdipisa) [inline docs/user-guide/geoprocessing-tool.md:28]:
```suggestion
The buffer layer is added in [TOC](toc.md) inside a new group created for the purpose.
```

File: docs/user-guide/geoprocessing-tool.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+### Advanced Settings
+
+Enabling the *Advanced options* the editor can include the following to the *Buffer* operation:
```
Comment (tdipisa) [inline docs/user-guide/geoprocessing-tool.md:32]:
```suggestion
Enabling the *Advanced options* the user can include further (not mandatory) parameters for the *Buffer* process:
```

File: docs/user-guide/geoprocessing-tool.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+<img src="../img/geoprocessing-tool/buffer-advanced-options.jpg" class="ms-docimage" width="400px" />
+
+* Enter the **Quadrant Segments**, that is the number of line segments used to approximate a quarter circle.
+
+* Select the **Style for the buffer end caps** choosing between `Round`, `Flat` or `Square`
+
+## Intersection tool
+
+Once the **Intersection** has been chosen as a process, the editor can select a layer to intersect from the **Source Layer** option drop down menu and one of the layer's features from the **Source Feature** option drop down menu or, enable the <img src="../img/button/add_marker_button.jpg" class="ms-docbutton"/> button that gives the possibility to select the *Feature* directly from the map.
```
Comment (tdipisa) [inline docs/user-guide/geoprocessing-tool.md:42]:
```suggestion
Once **Intersection** is selected as a process to be used from the first dropdown, the user can also select the layer to use as **Source Layer** for the intersection as well as the layer to be intersected with the given source (**Intersection Layer** option). For both options it is possible to select a feature to use for the intersection process; this is possible by enabling the <img src="../img/button/add_marker_button.jpg" class="ms-docbutton"/> buttons in order to select the layer features directly with a click on the map. If no layer feature is provided for one of the to layer options, the operation will be performed on the entire layer.
```

File: docs/user-guide/geoprocessing-tool.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+Once the **Intersection** has been chosen as a process, the editor can select a layer to intersect with the following operations:
```
Comment (tdipisa) [inline docs/user-guide/geoprocessing-tool.md:44]:
```suggestion
The user can then:
```

File: docs/user-guide/geoprocessing-tool.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+* Select the **Source Layer** from the drop down menu.
+
+* Select the layer's features from the **Source Feature**. The feature can be selected directly from the drop down menu or by clicking on the map, activating the <img src="../img/button/add_marker_button.jpg" class="ms-docbutton"/> button.
```
Comment (tdipisa) [inline docs/user-guide/geoprocessing-tool.md:48]:
```suggestion
* Select the layer feature from the **Source Feature**. The feature can be selected from the drop down menu or by clicking directly on the map by activating the <img src="../img/button/add_marker_button.jpg" class="ms-docbutton"/> button
```

File: docs/user-guide/geoprocessing-tool.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+<video class="ms-docimage" controls><source src="../img/geoprocessing-tool/select-feature.mp4"/></video>
+
+In the same way it is possible to select the **Intersection Layer** and the corresponding **Intersection Feature** to obtain the new intersected layer by clicking on <img src="../img/button/run_button.jpg" class="ms-docbutton"/> button.
+
+<video class="ms-docimage" controls><source src="../img/geoprocessing-tool/run_intersection-layer.mp4"/></video>
+
+The new layer, result of the intersection of the features selected, will be added to the [TOC](toc.md) and visible in the map viewer.
```
Comment (tdipisa) [inline docs/user-guide/geoprocessing-tool.md:56]:
```suggestion
The new layer, result of the intersection of the features selected, will be added to the [TOC](toc.md) inside a dedicated group and visible in the map viewer.
```

File: docs/user-guide/geoprocessing-tool.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+<img src="../img/geoprocessing-tool/intersection-layer.jpg" class="ms-docimage" />
+
+### Advanced Settings
+
+Enabling the *Advanced options* the editor can include the following to the *Intersection* operation:
```
Comment (tdipisa) [inline docs/user-guide/geoprocessing-tool.md:62]:
```suggestion
Enabling the *Advanced options* the user can include further (not mandatory) parameters for the *Intersection* process:
```

**Review by tdipisa — CHANGES_REQUESTED** [review]

File: docs/user-guide/geoprocessing-tool.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+
+The **Buffer** tool allows to create a buffer around the input geometries and when it is selected, the user can:
+
+* Select a layer from the **Source Layer** option drop down menu. The dropdown will show the layers available for the process.
+
+* Select one of the layer's features from the **Source Feature**. The feature can be selected directly by the dropdown menu or from the map, activating the <img src="../img/button/add_marker_button.jpg" class="ms-docbutton"/> button.
+
+<video class="ms-docimage" controls><source src="../img/geoprocessing-tool/select-buffer-feature.mp4"/></video>
+
+* Insert the desired **Distance** for the buffer.
```
Comment (tdipisa) [inline docs/user-guide/geoprocessing-tool.md:22]:
```suggestion
* Insert the desired **Distance** for the buffer (supported are _m_ and _km_).
```

**Review by tdipisa — CHANGES_REQUESTED** [review]

**Review by tdipisa — CHANGES_REQUESTED** [review]
@ElenaGallo the ESLint check is failing
https://github.com/geosolutions-it/MapStore2/actions/runs/6327934546/job/17184931983?pr=9496

**Comment by ElenaGallo** [comment]
Close in favor of #9505

**Pull request closed without merging**